src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -947,7 +952,8 @@ private bool CheckParameterValueKind(SyntaxNode node, BoundParameter parameter,
 
                 if (RequiresAssignableVariable(valueKind) && !backingField.ContainingType.IsReferenceType && (this.ContainingMemberOrLambda as MethodSymbol)?.IsEffectivelyReadOnly == true)
                 {
-                    ReportThisLvalueError(node, valueKind, isValueType: true, diagnostics);
+                    var errorCode = GetThisLvalueError(valueKind, isValueType: true);
```

Comment:
>GetThisLvalueError [](http://example.com/codeflow?start=36&length=18)

It looks like there are only two callers to `GetThisLvalueError()`. Given that, consider adding a `bool isPrimaryConstructorParameter` parameter, similar to the `bool isValueType` parameter, rather than checking the return value.